Essentials/src/main/java/com/earth2me/essentials/items/transform/PluginItemTransformer.java

```diff
@@ -0,0 +1,22 @@
+package com.earth2me.essentials.items.transform;
+
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.plugin.Plugin;
+
+public abstract class PluginItemTransformer {
+    private final Plugin plugin;
+
+    public PluginItemTransformer(final Plugin thePlugin) {
```

Comment:
It could be neat to provide plugins with a way to create a transformer using a lambda, but I wouldn't say this is required for this PR.